src/io/openshift/plugins/analytics.groovy

```diff
@@ -1,10 +1,27 @@
 package io.openshift.plugins
 
 import io.openshift.Events
+import io.openshift.Utils
 
 def register() {
     Events.on("build.pass") {
-        e, a -> echo "invoking bayesian analytics $e $a"
+        e, a -> echo "invoking bayesian analytics"
+        def image = config.runtime()
+        if(!image) {
+            return
+        }
+        try {
+            retry(3) {
+                def response = bayesianAnalysis(url: 'https://bayesian-link', gitUrl: a[0].git.url, ecosystem: image)
```

Comment:
@hrishin a[0] is quite ugly, we should replace this with something better like 



```groovy
class Events {

    static private listeners = [:]

    static def on(String event, Closure c) {
        on([event], c)
    }

    static def on(List events, Closure c) {
        events.each { e ->
            listeners[e] = listeners[e] ?: [] as Set
            listeners[e].add(c)
            println "registered for $e ${listeners[e]}"
        }
    }

    static def emit(String event, Object... args) {
        emit([event], args)
    }

    static def emit(List events, Object... args) {
        events.each { e ->
            if (!listeners[e]) {
                return
            }
            listeners[e].each { c -> c.call([name: e], *args) }
        }
    }

}



Events.on("build.pass") { e, a, ...rest ->
  println("foobar: event: $e | first: $a | rest: $rest")
  println("foobar: event: $e | first: $a.git.url")
}



Events.emit(["build.end", "build.pass"],
        [status: "status", namespace: "namespace",
         git: [url: "gitURL", commit: "commitHash"]])

```

Reply:
@hrishin the change being ` listeners[e].each { c -> c.call([name: e], *args) } `

Reply:
@hrishin https://github.com/fabric8io/osio-pipeline/pull/57

Reply:
thank you @sthaha for the fix. 

@yzainee could you please use `a.git.url` ?

Reply:
@yzainee instead of `a` please rename the var as `build` so that it reads `build.git.url`

```groovy

 Events.on("build.pass") { e, build -> 

 bayesianAnalysis(url: 'https://bayesian-link', gitUrl: build.git.url, ecosystem: image)

}
```

Reply:
@hrishin @sthaha Yes, renaming to build would be better.. but i cannot use build.git.url. I had already tried it. As its an array, it doesnt work without the indexing. Here is the exception which is thrown if i try without indexing

ERROR: Execution failed
org.jenkinsci.plugins.scriptsecurity.sandbox.RejectedAccessException: unclassified field java.lang.Object[] git
	at org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.SandboxInterceptor.unclassifiedField(SandboxInterceptor.java:397)
	at org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.SandboxInterceptor.onGetProperty(SandboxInterceptor.java:381)
	at org.kohsuke.groovy.sandbox.impl.Checker$6.call(Checker.java:288)
	at org.kohsuke.groovy.sandbox.impl.Checker.checkedGetProperty(Checker.java:292)
	at com.cloudbees.groovy.cps.sandbox.SandboxInvoker.getProperty(SandboxInvoker.java:29)
	at com.cloudbees.groovy.cps.impl.PropertyAccessBlock.rawGet(PropertyAccessBlock.java:20)
	at io.openshift.plugins.analytics.register(file:/var/lib/jenkins/jobs/yzainee/jobs/nodejs-health-check-redhat/branches/master/builds/13/libs/github.com/yzainee/osio-pipeline/src/io/openshift/plugins/analytics.groovy:20)

Reply:
@yzainee it seems to me that you may not have rebased onto latest master.